File: LeetCode/167-Two-Sum-II.cpp

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int high = numbers.size()-1;
        int low = 0;
        vector<int> ret;
        while(true){
            if(numbers[high] + numbers[low] < target){
                low++;
            }
            else if (numbers[high] + numbers[low] > target){
                high--;
            }
            else{
                ret.push_back(low+1);
                ret.push_back(high+1);
                return ret;
            }

        }

    }
};

File: LeetCode/167-Two-Sum-II.py
class Solution(object):
    def twoSum(self, nums, target):
        start = 0
        end = len(nums)-1
        while nums[start] + nums[end] != target:
            if nums[start] + nums[end] > target:
                end -= 1
            else:
                start +=1
        return [start+1, end+1]
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """